src/pages/index.js

```diff
@@ -27,71 +32,186 @@ const userinfoComponent = new Userinfo(
+      .finally(() => {
+        addCardPopup.hideLoading();
+      });
   },
-  body: JSON.stringify({
-    name: "Marie Skłodowska Curie",
-    about: "Physicist and Chemist",
-  }),
-});
-// Card
-function handleCardImageClick(cardData) {
-  imagePreviewModal.open(cardData);
-}
+  "Saving..."
+);
 
-function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleCardImageClick);
-  const cardElement = card.getView();
-  section.addItem(cardElement);
+// Delete Card //
+const deleteCardPopup = new PopupWithConfirmation(
+  "#delete-card-modal",
+  "Deleting..."
+);
+deleteCardPopup.setEventListeners();
+
+function handleDeleteClick(card, cardID) {
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.showLoading();
+    api
+      .deleteCard(cardID)
+      .then(() => {
+        card.removeCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => {
+        deleteCardPopup.hideLoading();
+      });
+  });
+
+  deleteCardPopup.open();
 }
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
+// Editing Profile //
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  (inputValues) => {
+    editAvatarPopup.showLoading();
+
+    api
+      .updateProfilePicture(inputValues)
+      .then(() => {
+        profileAvatar.src = inputValues.link;
+        editAvatarPopup.close();
+      })
+
+      .catch((err) => console.error(err))
+      .finally(() => {
+        editAvatarPopup.hideLoading();
+      });
   },
-  containerSelector
+  "Saving..."
 );
 
-section.renderItems();
+profileAvatarButton.addEventListener("click", () => {
+  editAvatarPopup.open();
+});
+
+editAvatarPopup.setEventListeners();
 
-// Modal Image
-const imagePreviewModal = new PopupWithImage(imageModalSelector);
-imagePreviewModal.setEventListeners();
+// Likes///
+function handleLikeClick(card) {
+  if (card.isLiked()) {
+    api
+      .likeCountRemove(card)
+      .then((updatedCard) => {
+        card.setLikes(updatedCard.likes);
+      })
+      .catch((error) => {
+        console.error("Error removing like:", error);
+      });
+  } else {
+    api
+      .likeCountAdd(card)
+      .then((updatedCard) => {
+        card.setLikes(updatedCard.likes);
+      })
+      .catch((error) => {
+        console.error("Error adding like:", error);
+      });
+  }
+}
+
+const section = new Section({}, containerSelector);
 
 // Functions
+
+function renderCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleCardImageClick,
+    (cardID) => handleDeleteClick(card, cardID),
+    handleLikeClick
+  );
+
+  return card.getView();
+}
+
 function addCard(data) {
   const cardData = {
     name: data.title,
     link: data.imageURL,
   };
 
-  renderCard(cardData);
+  const cardElement = renderCard(cardData);
+  section.addItem(cardElement);
+
   addCardFormPopup.close();
+
```

Comment:
You should start the loading before the request

Reply:
Please fix this issue

Reply:
Please fix this issue: start the loading using `renderLoading` method before the request